### PR TITLE
[SYCL][Matrix] Add generation of spirv.CooperativeMatrixKHR type

### PR DESCRIPTION
**Pull request by MrSidims**
Represented via target extension type. Joint matrix type will be removed soon.

### CODE REVIEW

**Comment by MrSidims** [comment]
It's a first part of https://github.com/intel/llvm/pull/13316

File: clang/lib/CodeGen/CodeGenTypes.cpp [hunk 1]
```diff
@@ -320,6 +320,23 @@ llvm::Type *getJointMatrixINTELExtType(llvm::Type *CompTy,
                                   "spirv.JointMatrixINTEL", {CompTy}, Params);
 }
 
+llvm::Type *
+getCooperativeMatrixKHRExtType(llvm::Type *CompTy,
+                               ArrayRef<TemplateArgument> TemplateArgs,
+                               const unsigned Val = 0) {
```
Comment (MrSidims) [inline clang/lib/CodeGen/CodeGenTypes.cpp:326]: Unused

**Comment by MrSidims** [comment]
@dpcpp-cfe-reviewers please take a look

**Review by elizabethandrews — APPROVED** [review]
Sorry for the delay. I missed the notifications on this

**Review by smanna12 — APPROVED** [review]

**Comment by MrSidims** [comment]
@intel/llvm-gatekeepers please help with the merge

**Pull request merged**